### ISSUE TITLE
GPGS crash when login

### ISSUE THREAD

**Issue by oniloures** [playgameservices/play-games-plugin-for-unity#517]
Hello!
I received this report from someone who bought my game (received vida Google Play Developer Console).

java.lang.Error: FATAL EXCEPTION [UnityMain]
Unity version : 4.5.5f1
Device model : HTC HTC One mini 2
Device fingerprint: htc/vodafone_uk/htc_memul:4.4.2/KOT49H/391137.2:user/release-keys

Caused by: java.lang.Error: signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 00000008
Build fingerprint: 'htc/vodafone_uk/htc_memul:4.4.2/KOT49H/391137.2:user/release-keys'
Revision: '0'
pid: 21950, tid: 22051, name: Thread-11928 >>> com.MarconiLoures.EarthlingsBeware <<<
r0 00000000 r1 a439f00d r2 4009138c r3 00000004
r4 00000011 r5 4c2d7ad0 r6 41785e00 r7 00000002
r8 4173cca4 r9 4c2f63f0 sl 00000000 fp 506dccb8
ip 506dcca8 sp 506dcb70 lr 416c4e15 pc 416995de cpsr 332f4839

at libart.001705de(Native Method)
at system@framework@boot.0001fefc(Native Method)
at NativeClient.HandleAuthTransition(Native Method)
at NativeClient.HandleAuthTransition(<0x001e8>:488)
at GameServicesBuilder.InternalAuthFinishedCallback(<0x00053>:83)
at GameServicesBuilder.InternalAuthFinishedCallback(Native Method)
at libgpg.000c130c(Native Method)
at libgpg.0005f1cd(Native Method)
at libgpg.0009f7c3(Native Method)
at libgpg.0005c3fd(Native Method)

He said:
"I downloaded it onto my HTC One M8 but now it keeps crashing after it tries to connect to Google play games." "It exits after right after it connects with Google play, the little thing at the top appears the error message is the show. I got through the menu When once there was a firewall blocking Google play (I think it was that) but que crashed the it was zooming in on the player both times I tried. Turing wifi and mobile data off does not solve it. "

The game stops in both parts where the GPGS checks if it is connected or trying to connect.
I didn't recognize anything in the code that can cause this. It only checks whether or not the player is connected to the GP.If not, it connects, if yes, the game goes on.

if (!Authenticated)
 {
PlayGamesPlatform.Activate();				
Social.localUser.Authenticate((bool success) =>{
if(success) { "the game goes on" }  
if(!success) { "show a error messenge and the game goes on without connect" }});
}
else { "the game goes on" }

Could someone confirm if it really is a problem in GPplugin (v0.9.15) or not?

**Comment by tyrmullen** [comment]
This is a strange crash, but looking at the stacktrace and the device fingerprint leads me to believe that the user who reported it is running Android KitKat (4.4.2) and has switched from the default Dalvik runtime to the experimental developer preview of ART.  Now (Android 5+) ART is the default, but old versions, and in particular the first highly experimental 4.4.2 release, have been known to exhibit undesirable behavior in certain cases (like crashes with a stack trace similar to the one above).

So TL;DR: doesn't sound like a problem in the plug-in, and if your user switches back to default Dalvik or updates Android, I would expect this crash to go away.

**Comment by oniloures** [comment]
Luckily, this guy contacted me. I can check with him and see if you solved this problem. I'll come back here and post the results.

Thank you very much for your attention!

**Comment by claywilkinson** [comment]
It's been a while with no activity.  @oniloures  - please open another issue if you are still experiencing problems.  Thanks!
